src/Native/System.Security.Cryptography.Native/pal_ecdsa_import_export.cpp

```diff
@@ -0,0 +1,461 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#include <openssl/ec.h>
+#include "pal_ecdsa_import_export.h"
+#include "pal_utilities.h"
+
+ECCurveType MethodToCurveType(EC_METHOD* method)
+{
+	if (method == EC_GFp_mont_method())
+		return ECCurveType::PrimeShortWeierstrass;// PrimeMontgomery;
+
+	if (method == EC_GF2m_simple_method())
+		return ECCurveType::Characteristic2;
+
+	// fallback including nist curves (EC_GFp_nist_method and others)
+	return ECCurveType::PrimeShortWeierstrass;
```

Comment:
Rather than be wrong, can we test these last ones and return an unknown when we don't know?